salt/minion.py

```diff
@@ -3242,3 +3242,67 @@ def _post_master_init(self, master):
         self.functions['saltutil.sync_grains'](saltenv='base')
         self.grains_cache = self.opts['grains']
         self.ready = True
+
+    @classmethod
+    def _target(cls, minion_instance, opts, data, connected):
+        if not minion_instance:
+            minion_instance = cls(opts)
+            minion_instance.connected = connected
+            if not hasattr(minion_instance, 'functions'):
+                # Need to load the modules so they get all the dunder variables
+                functions, returners, function_errors, executors = (
+                    minion_instance._load_modules(grains=opts['grains'])
+                    )
+                minion_instance.functions = functions
+                minion_instance.returners = returners
+                minion_instance.function_errors = function_errors
+                minion_instance.executors = executors
+
+                # we can then sync any proxymodules down from the master
+                # we do a sync_all here in case proxy code was installed by
+                # SPM or was manually placed in /srv/salt/_modules etc.
+                minion_instance.functions['saltutil.sync_all'](saltenv=minion_instance.opts['environment'])
```

Comment:
I think I can probably remove this.

Since we have already started the minion by the time we get here, the proxymodule should have already been available and synced down.